### DISCUSSION TITLE
hide page on dompdf based on if condition

### DISCUSSION THREAD

**Discussion by elsenthil123** [dompdf/dompdf#2766] [General]
Dear Team,

I generated a pdf report using DOMPDF. This pdf has multiple pages. I am using page break for all the pages. In this, I want to hide some pages based on my condition.

```
<main class=\"classname\">
  <header>
		<table>           
			
			<th1 ><img src='".$logo."' alt='".$logo."' style='width:180px;height:50px;' /><br /></th1>
			<th2 ></th2>           
			
			</table>
        </header>
 <h2 >Ref: SSI/$bnch/$date2/$id</h2>
 
<table width=\"100%\" border=\"0\" cellspacing=\"0\" cellpadding=\"3\" style=\"border:2px solid #000000\">
 <tr>
 <td class='head1'><b>Calibration Report for Aggregate Weigher</b></td>

 </tr>
 </table>
 
<table width=\"100%\" border= 2px solid #000000 cellspacing=\"0\" cellpadding=\"0\" style=\"border:2px solid #000000\" >
<thead>
<tr>
<td class='head2'><b>Ascending Order  </b></td>
<td class='head2'><b>Descending Order</b></td>
</tr>
</thead>
</table>


<table width=\"100%\"  border=\"1\" cellspacing=\"0\" cellpadding=\"1\" align=\"left\" style=\"border:2px solid #000000\"; >
<thead>
    <tr>
        <th class='head2'><b>Actual <br> Reading</b></th>
        <th class='head2'><b>Meter<br> Reading</b></th>
        <th class='head2'><b>Error</b></th>
        <th class='head2'><b>Actual <br> Reading</b></th>
        <th class='head2'><b>Meter<br> Reading</b></th>
        <th class='head2'><b>Error</b></th>
</tr>
</thead>
";

 $json_ascending = json_decode($aggregateascending, true);
  $json_dscending = json_decode($aggregatedescending, true);
  $asskey= array_keys($json_ascending);
  $deskey= array_keys($json_dscending); 
  $asscount = count($json_ascending);
  $dsscount = count($json_dscending);
  

      for ($i=0, $j=0; $i<$asscount; $i++,$j++){

      
      $ascending1 = explode(',', $json_ascending[$asskey[$i]]);
      $dscending1 = explode(',', $json_dscending[$deskey[$j]]);
$entry = " 

<tbody>
		<tr>
			<td class='head2'>$ascending1[0]</td>
			<td class='head2'>$ascending1[1]</td>
			<td class='head2'>$ascending1[2]</td>
			<td class='head2'>$dscending1[0]</td>
			<td class='head2'>$dscending1[1]</td>
			<td class='head2'>$dscending1[2]</td>
		</tr>
";
$html .=$entry;
		  }

$html1 = "     
 </tbody>
</table>


  <footer>
  <h2><picture>
      <img src='".$signature."' alt='".$signature."' style='width:150px;height:100px;' />
</picture></h2>
</footer>

 <div style='page-break-before:always'>&nbsp;</div>
 <div style='background-color: white; z-index: 2;'></div>
 
 </main>
```

The above code is one page. In that above code "$aggregateascending" is 0 then this page needs to be hidden. 

I was created style class 
```
.classname {
    display:none
}
```

When I am using this class the page is hiding. But it is not working with my condition.

Can anyone help me with how to use if condition for hiding this page? 

I am attaching my entire code here for your reference

[Code.txt](https://github.com/dompdf/dompdf/files/7984870/Code.txt)

**Comment by bsweeney** [comment]
It's easier for us to debug any issues with Dompdf if you can provide a sample of your document (HTML+CSS), rather than the PHP that generates the document.

**Reply by elsenthil123** [reply]
Hi bsweeny, 

Thanks for your reply. I am not using HTML. I am using PHP only

**Reply by bsweeney** [reply]
I understand that, but you are generating HTML + CSS for Dompdf. What would help us help you would be to see the value of $html that you are passing to Dompdf.

**Reply by elsenthil123** [reply]
I am not understand properly. Can you please explain me

**Reply by bsweeney** [reply]
It's difficult to debug an issue with how Dompdf renders a document when we can't see the document it's rendering. You have a lot of PHP code generating a document that you store in the $html variable. So, for us to determine what's going on we would need to see the value of the $html variable as it is when you pass it to Dompdf.

Near the end of your PHP script you have this line: `$dompdf->loadHtml($html);`. What is the value of the $html variable here?

**Reply by elsenthil123** [reply]
All my datas, which will display on pdf is defined inside the $htm only

**Reply by bsweeney** [reply]
Right. But I don't know what that looks like. You have loops and conditional logic and data pulled from the database. That's all irrelevant to what Dompdf is doing. The only thing Dompdf processes is HTML, and that's what you should share when you're asking for help debugging an issue with the PDF created by Dompdf.

**Reply by fd6130** [reply]
> All my datas, which will display on pdf is defined inside the $htm only

Provide samples such as screenshot of current generated html/pdf and the expected html/pdf is way better than just pure code.

**Reply by elsenthil123** [reply]
Dear Team,

I had uploaded my pdf with and without hiding the page. Please refer to the attachment.

For hiding the table I am just using

<main class=\"classname\">

<style> 
.classname { 
display:none 
} 
</style>
[pdf with hidden of Aggregate table.pdf](https://github.com/dompdf/dompdf/files/7991453/pdf.with.hidden.of.Aggregate.table.pdf)
[pdf without hidden of Aggregate table.pdf](https://github.com/dompdf/dompdf/files/7991454/pdf.without.hidden.of.Aggregate.table.pdf)

But I want it should work with if condition

**Reply by bsweeney** [reply]
Is this a programming question? Are you asking how to conditionally add the "if" clause to your PHP? You may have better luck on a site like StackOverflow. Not that we can't help but you'll get more visibility for your question on a site like that.

Or, do you have some structural condition that would determine when the table is hidden? Looking for a solution based on the document structure/styling?

**Comment by markkimsal** [comment]
It might be easier to render each page as a separate PDF and then use FPDI to glue them together into 1 final document using `importPage()`
